units/src/amount.rs

```diff
@@ -1155,10 +1165,27 @@ impl ops::DivAssign<u64> for Amount {
     fn div_assign(&mut self, rhs: u64) { *self = *self / rhs }
 }
 
+#[cfg(feature = "alloc")]
 impl FromStr for Amount {
     type Err = ParseError;
 
-    fn from_str(s: &str) -> Result<Self, Self::Err> { Amount::from_str_with_denomination(s) }
+    fn from_str(s: &str) -> Result<Self, Self::Err> {
+        let a = Amount::from_str_with_denomination(s);
```

Comment:
Please name the variable `result`.

Reply:
result seems kinda verbose, why not just r?  Anyway not worth nit picking about so I've named it result.